### ISSUE TITLE
Qwen3-emebdding训练中卡在eval不动

### ISSUE THREAD

**Issue by HaimianYu** [QwenLM/Qwen3-Embedding#108]
单卡跑的时候正常，多卡跑会卡在evaluation那里。transformers为=4.51.3，ms-swift=3.6.1
下面是运行脚本：

`model_path=xxx
train_data=data/qwen3_embed/demo.jsonl
output_dir=checkpoints/qwen3_embedding/v1_train
log_path=log/qwen3_embedding_v1_train.log
batch_size=24

export INFONCE_USE_BATCH=False

export MASTER_PORT=29501
nproc_per_node=4
NPROC_PER_NODE=$nproc_per_node
CUDA_VISIBLE_DEVICES=0,1,2,3 nohup swift sft
--model $model_path
--task_type embedding
--model_type qwen3_emb
--train_type full
--dataset $train_data
--split_dataset_ratio 0.05
--eval_strategy steps
--output_dir $output_dir
--eval_steps 10
--num_train_epochs 5
--save_steps 10
--per_device_train_batch_size $batch_size
--per_device_eval_batch_size $batch_size
--gradient_accumulation_steps 1
--learning_rate 6e-6
--loss_type infonce
--label_names labels
--dataloader_drop_last true
--deepspeed zero3 &>$log_path &`

**Comment by realqnn** [comment]
同样的问题
